optuna/storages/_in_memory.py

```diff
@@ -362,14 +360,12 @@ def get_all_trials(
         with self._lock:
             self._check_study_id(study_id)
 
-            trials: Union[List[FrozenTrial], Iterator[FrozenTrial]] = self._studies[
-                study_id
-            ].trials
+            trials = self._studies[study_id].trials
             if states is not None:
-                trials = filter(lambda t: t.state in states, trials)
+                trials = [t for t in trials if t.state in states]
 
             if deepcopy:
-                trials = copy.deepcopy(list(trials))
+                trials = copy.deepcopy(trials)
             else:
                 trials = list(trials)
```

Comment:
[nit] Can we also remove the call of `list()` here since `self._studies[...].trials` is `List[FrozenTrial]`?

```suggestion
```

Reply:
This shallow copy is necessary to ensure that returned trials are not modified by `set_trial_xxx`. I added commit for clarity.

Reply:
I see. Thank you for the update!